### PR TITLE
Add support for ODROID-GO Game Kit

### PR DESCRIPTION
**Pull request by ydamigos**
Add support for ODROID-GO Game Kit

Tested samples:
```
samples/hello_world
samples/basic/blinky
```

### CODE REVIEW

File: boards/xtensa/odroid_go/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+Programming
+***********
+
+Two components are required in order to build applications for this board: the
+`toolchain`_ and the `SDK`_.  Both are provided by the SoC manufacturer.
+
+The SDK contains headers and a hardware abstraction layer library (provided only
+as object files) that are required for the port to function.
+
+The toolchain is available for Linux, Windows, and Mac hosts and instructions to
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]: Let's expose these links better (and fix a typo) with:
```
The toolchain is available for Linux, Windows, and macOS hosts and instructions to
obtain and set them up are available in the `ESP-IDF repository
<https://github.com/espressif/esp-idf>`_, as explained in the
`ESP-IDF Programming Guide <SDK>`_.
```

File: boards/xtensa/odroid_go/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   export ZEPHYR_TOOLCHAIN_VARIANT="espressif"
+   export ESPRESSIF_TOOLCHAIN_PATH="/path/to/xtensa-esp32-elf/"
+
+Flashing
+--------
+
+The usual ``flash`` target will work with the ``odroid_go`` board configuration.
+Here is an example for the :ref:`hello_world` application using CMake:
+
+.. code-block:: console
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]: Can we use the ``.. zephyr-app-commands`` directive here (see samples/hello_world/README.rst for an example use).

File: boards/xtensa/odroid_go/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+For example, to flash to ``/dev/ttyUSB0``, use the following command after
+having build the application in the ``build`` directory:
+
+.. code-block:: console
+
+   west flash -d build/ --skip-rebuild --esp-tool=/path/to/esp-idf/components/esptool_py/esptool/esptool.py --esp-device /dev/ttyUSB0
+
+References
+**********
+
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]: If you want the toolchain and SDK links (and not just the footnote) to show up here too, add the directive (followed by a blank line):
```
.. target-notes::

```

File: boards/xtensa/odroid_go/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                        serial port to flash, default /dev/ttyUSB0
+  --esp-baud-rate ESP_BAUD_RATE
+                        serial baud rate, default 921600
+  --esp-flash-size ESP_FLASH_SIZE
+                        flash size, default "detect"
+  --esp-flash-freq ESP_FLASH_FREQ
+                        flash frequency, default "40m"
+  --esp-flash-mode ESP_FLASH_MODE
+                        flash mode, default "dio"
+  --esp-tool ESP_TOOL   if given, complete path to espidf. default is to
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]: probably don't need the phrase ``if given,  ``

File: boards/xtensa/odroid_go/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        # Use cmake to configure buildsystem:
+        cmake -DESP_IDF_PATH=/path/to/esp-idf/ -B build -DBOARD=odroid_go samples/hello_world
+
+        # Now run make to build the application:
+        make
+
+Refer to :ref:`build_an_application` and :ref:`application_run` for more
+details.
+
+All flashing options are now handled by the :ref:`west` tool, including flashing
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]:
```suggestion
All flashing options are handled by the :ref:`west` tool, including flashing
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/xtensa/odroid_go/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  --esp-flash-partition_table ESP_FLASH_PARTITION_TABLE
+                        Partition table to flash
+
+For example, to flash to ``/dev/ttyUSB0``, use the following command after
+having build the application in the ``build`` directory:
+
+.. code-block:: console
+
+   west flash -d build/ --skip-rebuild --esp-tool=/path/to/esp-idf/components/esptool_py/esptool/esptool.py --esp-device /dev/ttyUSB0
+
```
Comment (dbkinder) [inline boards/xtensa/odroid_go/doc/index.rst]: Can you also add what happens when you run the hello_world sample?  I doubt it displays the usual "hello world" message on the screen, so do you need instructions that hook up a serial console to see that?  (Or are you relying on the link to application_run to cover that?)

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by ydamigos** [comment]
@dbkinder Thanks for reviewing. All comments addressed.

**Review by dbkinder — APPROVED** [review]
docs LGTM, thanks

**Review by dcpleung — APPROVED** [review]

**Comment by ydamigos** [comment]
Rebased to master and remove `version` from `espressif,esp32-gpio.yaml`.

**Comment by ydamigos** [comment]
Rebased to master.

**Comment by carlescufi** [comment]
@galak could you take a look please?

**Comment by ydamigos** [comment]
Rebased to master and updated board's device tree for the new UART driver.

**Comment by ydamigos** [comment]
Rebased to master.

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**